features/collection/loaders/index.ts

```diff
@@ -0,0 +1,150 @@
+      collectionWithMetafieldsQuery,
+      {
+         handle,
+      }
+   );
+   if (response.data.collectionByHandle == null) {
+      return null;
+   }
+   const hierarchyMetafield = response.data.collectionByHandle.metafields.edges.find(
+      (edge) => edge.node.key === 'collection_hierarchy'
+   );
+   let ancestors: Collection[] = [];
+   let children: Collection[] = [];
+   if (hierarchyMetafield) {
+      const hierarchy = JSON.parse(hierarchyMetafield.node.value);
+      const childrenHandles = getChildrenHandles(hierarchy);
+      children = await loadCollections(childrenHandles, client);
+      const ancestorsHandles = getAncestorHandles(hierarchy).reverse();
+      ancestors = await loadCollections(ancestorsHandles, client);
+   }
+   return {
+      handle,
+      title: response.data.collectionByHandle.title,
+      description: response.data.collectionByHandle.description,
+      image: response.data.collectionByHandle.image,
+      ancestors,
+      children,
+   };
+}
+
+async function loadCollections(
+   handles: string[],
+   client: StorefrontClient
+): Promise<Collection[]> {
+   const responses = await Promise.all(
+      handles.map((handle) => client.request<Data>(collectionQuery, { handle }))
+   );
+   return responses
+      .filter((response) => response.data.collectionByHandle != null)
+      .map((response) => {
+         return {
+            handle: response.data.collectionByHandle.handle,
+            title: response.data.collectionByHandle.title,
+            image: response.data.collectionByHandle.image,
+            ancestors: [],
+            children: [],
+         };
+      });
+}
+
+function getChildrenHandles(metafieldParsedValue: any): string[] {
+   if (Array.isArray(metafieldParsedValue.children)) {
+      return metafieldParsedValue.children.map((handle) =>
+         handle.toLowerCase()
+      );
+   }
+   return [];
+}
+
+function getAncestorHandles(metafieldParsedValue: any): string[] {
+   if (Array.isArray(metafieldParsedValue.ancestors)) {
+      return metafieldParsedValue.ancestors.map((handle) =>
+         handle.toLowerCase()
+      );
+   }
+   return [];
+}
+
+type DataWithMetafields = {
+   data: {
+      collectionByHandle: {
+         handle: string;
+         title: string;
+         description: string;
+         image?: {
+            alt: string;
+            url: string;
+         };
+         metafields: {
+            edges: Array<{
+               node: {
+                  key: string;
+                  value: string;
+               };
+            }>;
+         };
+      };
+   };
+};
+
+const collectionWithMetafieldsQuery = /* GraphQL */ `
+   query collection($handle: String!) {
+      collectionByHandle(handle: $handle) {
+         handle
+         title
+         description
+         image {
+            alt: altText
+            url: transformedSrc
+         }
+         metafields(first: 10, namespace: "ifixit") {
```

Comment:
We might, but not right now. Keep in mind that when I wrote this I though of it as equivalent to mocking, but we real data. I haven't really carefully considered optimizations, because at this stage I'm focused on the collection UX/UI part. Once it is clear what the production environment is, then we can focus on optimizing API calls (for example, on your US store you might want to fetch data from your apis instead of Shopify).